### PR TITLE
Add XDG_CONFIG_HOME support

### PR DESCRIPTION
**Pull request by yemouu**
Signed-off-by: yemou <meeyou@tuta.io>

### CODE REVIEW

File: src/util/wf-shell-app.cpp [hunk 1]
```diff
@@ -10,9 +10,13 @@
 
 std::string WayfireShellApp::get_config_file()
 {
-    std::string home_dir = getenv("HOME");
-    std::string config_file = home_dir + "/.config/wf-shell.ini";
-    return config_file;
+    std::string config_dir = getenv("XDG_CONFIG_HOME");
```
Comment (ammen99) [inline src/util/wf-shell-app.cpp:13]: The old code has this bug as well, but we'd better fix it while we're at it: getenv returns `NULL` not an empty string when the environment variable is not found. So we cannot assign the result directly to a `std::string`, we first need to store the result in a `char*`, compare with null, and then decide.

**Comment by yemouu** [comment]
Will this work? I think it would be safe to assume that HOME is set.

**Review by ammen99 — APPROVED** [review]
LGTM, thanks!

**Comment by ammen99** [comment]
> Will this work? I think it would be safe to assume that HOME is set.

Yes, indeed, HOME is a rather safe assumption .. Ofc we'll crash without it set but I can't imagine why the user wouldn't have this.

**Pull request merged**